tuner/tuner/candidate_gen.py

```diff
@@ -162,6 +172,44 @@ def apply_params(
         )
         return MLIRTransformation(template, modified, embeddable)
 
+    def get_reduction_tile_sizes(self, configuration: Configuration):
+        tile_sizes = [0, 0]
+        tile_sizes.append(configuration.tile_sizes[2])
+        return tile_sizes
+
+    def get_workgroup_tile_sizes(self, configuration: Configuration):
+        tile_sizes = configuration.tile_sizes[:2]
+        tile_sizes.append(0)
+        return tile_sizes
+
+    def get_td_spec(
+        self,
+        ir_module: ir.Module,
+        configuration: Configuration,
+    ) -> ir.Module:
+        mmt_op: ir.Operation = self.get_mmt_operation(ir_module)
+        reduction_tile_sizes = self.get_reduction_tile_sizes(configuration)
+        workgroup_tile_sizes = self.get_workgroup_tile_sizes(configuration)
+        lowering_config = build_vector_distribute_lowering_config(
+            configuration,
+            reduction_tile_sizes,
+            workgroup_tile_sizes,
+            ir_module.context,
+        )
+        translation_info = build_vector_distribute_translation_info(
+            configuration, ir_module.context
+        )
+        lhs_type = ir.ShapedType(mmt_op.operands[0].type)
+        rhs_type = ir.ShapedType(mmt_op.operands[1].type)
+        acc_type = ir.ShapedType(mmt_op.operands[2].type)
+        M = acc_type.get_dim_size(0)
+        N = acc_type.get_dim_size(1)
+        K = lhs_type.get_dim_size(1)
+        func_name = f"match_mmt_{M}x{N}x{K}_{lhs_type.element_type}x{rhs_type.element_type}x{acc_type.element_type}"
+        return build_td_spec(
```

Comment:
I wonder if we should capture the exact function name in the tuner / problemsize? I don't think we should be able to always 'guess it', assuming future compiler changes